### PR TITLE
Unicode RL1.5 support: Implement Case-Insensitive Matching Mode

### PR DESCRIPTION
**Pull request by yuuki14202028**
## Summary

This PR implements case-insensitive matching support using the `(?i)` inline flag modifier, addressing #140.

## Changes

- Added `(?i)` and `(?-i)` flag modifier parsing in `Regex/Syntax/Parser/Basic.lean`
- Introduced `Ast.flags` node to represent inline flag modifiers in the AST
- Refactored `Ast.toRegexAux` from index-based to state-based transformation using `ToRegexState` to track both group index and case-insensitivity flag
- Created `Regex/Unicode/CaseFold.lean` implementing Unicode Simple Case Folding (status C+S from UCD CaseFolding.txt)
- Added `getCaseFoldEquivChars` to generate character equivalence classes for case-insensitive matching
- Included Unicode 17.0 Simple Case Folding data in `regex/data/`

## Implementation Notes

- Case-insensitive matching expands each character to an alternation of its case-equivalent characters (e.g., `(?i)a` becomes `a|A`)
- Uses Simple Case Folding for single-character mappings only (Full Case Folding with multi-character expansions like ß→ss is not supported)
- Flag scope follows standard regex semantics: `(?i)` enables and `(?-i)` disables case-insensitivity for subsequent patterns

## Todo

- [x] Added parser tests for `(?i)` and `(?-i)` syntax in `Parser/Test.lean`
- [x] Support combining flags with non-capturing groups (e.g., `(?i:abc)`)
- [x] Add edge case tests for multi-character case equivalence (e.g., Σ/σ/ς, K/k/K)
- [x] Add generate script Simple_Case_Folding.txt with .lean 
- [x] Add `caseInsensitive` option to `Regex.parse` and `Regex.parse!` methods
- [x] Update `correctness/` proofs to accommodate `Ast.flags` and `ToRegexState` changes
- [ ] (Optional) Support case-insensitive matching for character class ranges (e.g., `(?i)[a-z]`)

### CODE REVIEW

File: regex/Regex/Test.lean [hunk 1]
```diff
@@ -0,0 +1,27 @@
+-- match tests for the regex.
```
Comment (pandaman64) [inline regex/Regex/Test.lean]: `regex/tests/Test.lean`にテスト用ファイルがあります。

File: regex/Regex/Unicode/CaseFold.lean [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    | some arr => result := result.insert tgt (arr.push src)
+    | none => result := result.insert tgt #[tgt, src]
+  return result
+
+private def caseFoldEquivTableThunk : Thunk (Std.HashMap UInt32 (Array UInt32)) :=
+  Thunk.mk fun _ => buildCaseFoldEquivTable
+
+private def caseFoldEquivTable : Std.HashMap UInt32 (Array UInt32) := caseFoldEquivTableThunk.get
+
+def getCaseFoldEquivChars (c : Char) : Array Char :=
```
Comment (pandaman64) [inline regex/Regex/Unicode/CaseFold.lean]: この処理が正しいことを証明するのに興味はありますか？

**Review by pandaman64 — CHANGES_REQUESTED** [review]
ありがとうございます！

まだドラフトですがいくつかコメントしました。

File: regex/data/Simple_Case_Folding.txt [hunk 3]
```diff
@@ -0,0 +1,1512 @@
+0041;0061
```
[the remark this answers is not in the document]
Comment (yuuki14202028) [reply regex/data/Simple_Case_Folding.txt]: Bashで適当に生成しました。
了解です。Leanで書いてみます。

File: regex/Regex/Unicode/CaseFold.lean (hunk 2 above)
Comment (yuuki14202028) [reply regex/Regex/Unicode/CaseFold.lean]: 証明を書いてみることに興味はあります。
この場合は、何を証明するべきですかね？

Comment (yuuki14202028) [reply regex/Regex/Unicode/CaseFold.lean]:
```lean4
def CaseFoldEquiv (c₁ c₂ : Char) : Prop :=
  c₂ ∈ Internal.getCaseFoldEquivChars_spec c₁
...
theorem CaseFoldEquiv.equivalence : Equivalence CaseFoldEquiv :=
  ⟨CaseFoldEquiv.refl, CaseFoldEquiv.symm, CaseFoldEquiv.trans⟩
```
ASCII用に高速化してある部分とcaseFoldTableとの整合性を証明することが難しかったので、caseFoldTableが理想的なデータだと仮定した時に考えられるgetCaseFoldEquivChars_spec, getCaseFoldChar_specに対して、同値関係の証明を行いました。
ご確認ください。
